test-framework/src/main/java/org/elasticsearch/test/TestSearchContext.java

```diff
@@ -549,18 +549,11 @@ public FetchSearchResult fetchResult() {
         return null;
     }
 
-    @Override
-    public MappedFieldType smartNameFieldType(String name) {
-        if (mapperService() != null) {
-            return mapperService().smartNameFieldType(name, types());
-        }
-        return null;
-    }
 
     @Override
-    public MappedFieldType smartNameFieldTypeFromAnyType(String name) {
+    public MappedFieldType smartNameFieldType(String name) {
```

Comment:
We should probably rename this method in SearchContext since it is no longer "smart"? Just an idea for a followup.